pkg/controller/csr/csr_controller.go

```diff
@@ -53,17 +53,38 @@ func validUsername(csr *certificatesv1.CertificateSigningRequest, clusterName st
 	return csr.Spec.Username == fmt.Sprintf(userNameSignature, clusterName, clusterName)
 }
 
-func csrPredicate(csr *certificatesv1.CertificateSigningRequest) bool {
+// isValidUnapprovedBootstrapCSR checks if the CSR:
+// 1. Has a non-empty cluster name label
+// 2. Has not been approved or denied
+func isValidUnapprovedBootstrapCSR(csr *certificatesv1.CertificateSigningRequest) bool {
 	clusterName := getClusterName(csr)
 	return clusterName != "" &&
-		getApprovalType(csr) == "" &&
-		validUsername(csr, clusterName)
+		getApprovalType(csr) == ""
+}
+
+// approveExistingManagedClusterCSR checks if the CSR is from an existing managed cluster
+func approveExistingManagedClusterCSR(ctx context.Context, csr *certificatesv1.CertificateSigningRequest,
+	clientHolder *helpers.ClientHolder) (bool, error) {
+	if !validUsername(csr, getClusterName(csr)) {
```

Comment:
@zhujian7 The `validUsername` is put into the default approveCondition, so old logic still work.